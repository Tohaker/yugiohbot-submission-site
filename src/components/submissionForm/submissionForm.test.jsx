import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";

describe("Submission Form", () => {
  let SubmissionForm;
  let wrapper;
  const mockSubmitForm = jest.fn();

  beforeEach(() => {
    jest.mock("api/storage", () => ({ submitForm: mockSubmitForm }));

    SubmissionForm = require(".").default;
    wrapper = mount(<SubmissionForm />);
  });

  it("should render correctly", () => {
    snapshot(wrapper);
  });

  describe("When the submit button is pressed", () => {
    describe("With all fields filled", () => {
      it("should call submitForm with all fields", () => {
        act(() => {
          wrapper
            .find("input")
            .at(0)
            .simulate("change", {
              target: { value: "testTitle" }
            });
          wrapper
            .find("input")
            .at(1)
            .simulate("change", {
              target: { files: ["testImage.jpg"] }
            });
          wrapper
            .find("input")
            .at(2)
            .simulate("change", { target: { value: "testEffect" } });
        });

        act(() => {
          wrapper.find("button").simulate("click");
        });

        expect(mockSubmitForm).toHaveBeenCalledWith(
          "testImage.jpg",
          "testTitle",
          "testEffect"
        );
      });
    });
  });
});
