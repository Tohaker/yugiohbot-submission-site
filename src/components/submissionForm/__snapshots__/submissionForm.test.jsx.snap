// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submission Form should render correctly 1`] = `
<SubmissionForm>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "qpQpp",
            "rules": Array [
              "
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  text-align: left;
  color: white;
  padding: 0 8px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa qpQpp"
      >
        <label>
          Card Title:
        </label>
        <styled.input
          name="title"
          onChange={[Function]}
          type="text"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "iFxOLg",
                  "rules": Array [
                    "
  height: calc(10px + 3vmin);
  font-size: calc(10px + 2vmin);
  margin: 1vmin 0 3vmin 0;

  input {
    background-color: white;
  }
",
                  ],
                },
                "displayName": "styled.input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            name="title"
            onChange={[Function]}
            type="text"
          >
            <input
              className="sc-bwzfXH iFxOLg"
              name="title"
              onChange={[Function]}
              type="text"
            />
          </StyledComponent>
        </styled.input>
        <label>
          Card Image:
        </label>
        <styled.input
          name="image"
          onChange={[Function]}
          type="file"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "iFxOLg",
                  "rules": Array [
                    "
  height: calc(10px + 3vmin);
  font-size: calc(10px + 2vmin);
  margin: 1vmin 0 3vmin 0;

  input {
    background-color: white;
  }
",
                  ],
                },
                "displayName": "styled.input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            name="image"
            onChange={[Function]}
            type="file"
          >
            <input
              className="sc-bwzfXH iFxOLg"
              name="image"
              onChange={[Function]}
              type="file"
            />
          </StyledComponent>
        </styled.input>
        <label>
          Card Effect:
        </label>
        <styled.input
          name="effect"
          onChange={[Function]}
          type="text"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "iFxOLg",
                  "rules": Array [
                    "
  height: calc(10px + 3vmin);
  font-size: calc(10px + 2vmin);
  margin: 1vmin 0 3vmin 0;

  input {
    background-color: white;
  }
",
                  ],
                },
                "displayName": "styled.input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            name="effect"
            onChange={[Function]}
            type="text"
          >
            <input
              className="sc-bwzfXH iFxOLg"
              name="effect"
              onChange={[Function]}
              type="text"
            />
          </StyledComponent>
        </styled.input>
        <styled.button
          disabled={false}
          id="submit"
          onClick={[Function]}
        >
          <StyledComponent
            disabled={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "bAkHCv",
                  "rules": Array [
                    "
  font-size: calc(10px + 2vmin);
  background-color: white;
  border-radius: 50px;
",
                  ],
                },
                "displayName": "styled.button",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            id="submit"
            onClick={[Function]}
          >
            <button
              className="sc-htpNat bAkHCv"
              disabled={false}
              id="submit"
              onClick={[Function]}
            >
              Submit
            </button>
          </StyledComponent>
        </styled.button>
        <label />
      </div>
    </StyledComponent>
  </styled.div>
</SubmissionForm>
`;
